File: src/mlogix/mlogix/type/TypeRegistry.java
package mlogix.mlogix.type;

import java.util.*;

public class TypeRegistry {
    private final Map<String, Type> types = new HashMap<>();
    
    public TypeRegistry() {
        register(BuiltinType.Int);
        register(BuiltinType.Str);
    }
    
    public void register(Type type) {
        types.put(type.name, type);
    }
    
    public Type get(String name) {
        return types.get(name);
    }
    
    public boolean containsKey(String name) {
        return types.containsKey(name);
    }
}

File: src/mlogix/compiler/type/TypeRegistry.kt
package mlogix.compiler.type

import arc.struct.ObjectMap

class TypeRegistry {
    private val types: ObjectMap<String, Type> = ObjectMap<String, Type>()

    init {
        register(BuiltinType.Int)
        register(BuiltinType.Str)
    }

    fun register(type: Type) {
        types.put(type.name, type)
    }

    fun get(name: String?): Type? {
        return types.get(name)
    }

    fun containsKey(name: String?): Boolean {
        return types.containsKey(name)
    }
}